File: app/src/test/java/de/jodamob/kotlin/droidcon/f/VeryImportantFragmentTest.java
package de.jodamob.kotlin.droidcon.f;

import android.os.Bundle;

import org.junit.Test;
import org.mockito.InOrder;

import static junit.framework.Assert.assertEquals;
import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.inOrder;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class VeryImportantFragmentTest {

    Bundle bundle = mock(Bundle.class);
    VeryImportantFragment tested = new VeryImportantFragment();

    @Test
    public void readFromBundle() {
        when(bundle.getString(anyString())).thenReturn("this is a test");
        tested.onCreate(bundle);
        verify(bundle).getString("secretKey1");
        assertEquals("this is a test", tested.getImportantState());
    }

    @Test
    public void writesToBundle() {
        InOrder verifier = inOrder(bundle);

        tested.onSaveInstanceState(bundle);

        verifier.verify(bundle).putString("secretKey1", "i am most important");
        verifier.verify(bundle).putInt("secretKey2", 2);
        verifier.verifyNoMoreInteractions();
    }

}

File: app/src/test/java/de/jodamob/kotlin/droidcon/f/VeryImportantFragmentTest.kt
package de.jodamob.kotlin.droidcon.f

import android.os.Bundle
import com.nhaarman.mockito_kotlin.doReturn
import com.nhaarman.mockito_kotlin.inOrder
import com.nhaarman.mockito_kotlin.mock
import org.amshove.kluent.*
import org.junit.Test
import org.mockito.ArgumentMatchers.anyString

class VeryImportantFragmentTest {

    val bundle = mock<Bundle> {
        on {getString(anyString())} doReturn "test"
    }
    val tested = VeryImportantFragment()

    @Test
    fun readsFromBundle() {
        tested.onCreate(bundle)

        Verify on bundle that bundle.getString("secretKey1") was called
        tested.importantState `should equal` "test"
    }

    @Test
    fun writesToBundle() {
        tested.onSaveInstanceState(bundle)

        inOrder(bundle) {
            verify(bundle).putString("secretKey1", "i am most important")
            verify(bundle).putInt("secretKey2", 2)
            verifyNoMoreInteractions()
        }
    }

}